c/driver/common/utils.c

```diff
@@ -138,11 +138,11 @@ int StringBuilderAppend(struct StringBuilder* builder, const char* fmt, ...) {
   va_end(argptr);
 
   if (n < 0) {
-    return -1;
+    return errno;
```

Comment:
From man page, vsnprintf won't set errno so we should return our own value (EIO?)

Reply:
Hmm really? Trying to find a definitive list of library calls that do or don't set errno without luck, but as far as I can tell in POSIX (through clicking all the references in https://pubs.opengroup.org/onlinepubs/7908799/xsh/vsprintf.html) this has similar error handling to fputc which does explicitly set errno https://pubs.opengroup.org/onlinepubs/7908799/xsh/fputc.html

Microsoft documentation is thankfully more explicit:

https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/vsnprintf-vsnprintf-vsnprintf-l-vsnwprintf-vsnwprintf-l?view=msvc-170#return-value

Reply:
Hmm, my manpage on macOS doesn't mention errno at all.

Ah, but it does mention specific _codes_ without explicitly mentioning errno. OK, this is fine then. Sorry!